### PR TITLE
aw-awatcher: init at 0.2.7

### PR DESCRIPTION
**Pull request by BhasherBEL**
## Description of changes
https://github.com/2e3s/awatcher

 Activity and idle watchers for activitywatch compatible with wayland and hyprland.

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/aw/aw-awatcher/package.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    outputHashes = {
+      "aw-client-rust-0.1.0" = "sha256-fCjVfmjrwMSa8MFgnC8n5jPzdaqSmNNdMRaYHNbs8Bo=";
+    };
+  };
+
+  nativeBuildInputs = [ pkg-config ];
+
+  buildInputs = [ openssl ];
+
+  meta = with lib; {
```
Comment (0x61nas) [inline pkgs/by-name/aw/aw-awatcher/package.nix:30]: could you avoid using `with lib` here because it is a problematic anti-pattern, refer to: https://github.com/NixOS/nixpkgs/issues/208242

File: pkgs/by-name/aw/aw-awatcher/Cargo.lock
Comment (0x61nas) [inline pkgs/by-name/aw/aw-awatcher/Cargo.lock:1]: the upstream have [`Cargo.lock`](https://github.com/2e3s/awatcher/blob/main/Cargo.lock) already, why do we need to maintain a our own one?

**Review by 0x61nas — CHANGES_REQUESTED** [review]

File: pkgs/by-name/aw/aw-awatcher/Cargo.lock
Comment (BhasherBEL) [reply pkgs/by-name/aw/aw-awatcher/Cargo.lock:1]: Because based on [the documentation](https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/rust.section.md) and my experiments, it's not working as there is Cargo `git` dependencies.

Comment (0x61nas) [reply pkgs/by-name/aw/aw-awatcher/Cargo.lock:1]: you can achieve the same thing by:

```nix
  cargoLock = {
    lockFile = src + /Cargo.lock;
    outputHashes = {
      "aw-client-rust-0.1.0" = "sha256-fCjVfmjrwMSa8MFgnC8n5jPzdaqSmNNdMRaYHNbs8Bo=";
    };
  };
```

by doing this you'll use `Cargo.lock` from the upstream source and you can provide the git deps hashes. refer to: https://github.com/NixOS/nixpkgs/pull/316252/files#diff-8ac422c1d925671c1049e51f44d5fe3338fce5e9e1dce3fa0e7d1f6e3cfe1425R28

File: pkgs/by-name/aw/aw-awatcher/package.nix [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+  nativeBuildInputs = [ pkg-config ];
+
+  buildInputs = [ openssl ];
+
+  meta = {
+    description = "Activity and idle watchers";
+    homepage = "https://github.com/2e3s/awatcher";
+    license = lib.licenses.mpl20;
+    maintainers = [ lib.maintainers.bhasher ];
```
Comment (0x61nas) [inline pkgs/by-name/aw/aw-awatcher/package.nix:34]:
```suggestion
    maintainers = with lib.maintainers; [ bhasher ];
```

File: pkgs/by-name/aw/aw-awatcher/package.nix [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+  buildInputs = [ openssl ];
+
+  meta = {
+    description = "Activity and idle watchers";
+    homepage = "https://github.com/2e3s/awatcher";
+    license = lib.licenses.mpl20;
+    maintainers = with lib.maintainers; [ bhasher ];
+    mainProgram = "awatcher";
+    platforms = lib.platforms.linux;
```
Comment (0x61nas) [inline pkgs/by-name/aw/aw-awatcher/package.nix:36]: am not sure if we need to mark it as broken on `darwin` if we already exclude it here. but I suppose it's not required by this.

**Review by 0x61nas — APPROVED** [review]
works for me, thanks

**Comment by sifmelcara** [comment]
This is duplicate of #287858 (which is stalled), also here is a reference to the packaging request: #287826

@aikooo7 are you still interested in merging the PR?

**Comment by aikooo7** [comment]
> This is duplicate of #287858 (which is stalled), also here is a reference to the packaging request: #287826
> 
> @aikooo7 are you still interested in merging the PR?

My bad for being so much time out first of all.

I could fix it and merge it but since it is done here and people say it works we should perhaps merge this one?

File: maintainers/maintainer-list.nix
Comment (aikooo7) [inline maintainers/maintainer-list.nix:1]: Shouldn't this be in a separate commit?

**Comment by aikooo7** [comment]
> This is duplicate of #287858 (which is stalled), also here is a reference to the packaging request: #287826
> 
> @aikooo7 are you still interested in merging the PR?

I had indeed some stuff to do in that pr that I didn't see, my bad. Everything should be fixed now and now I am also using the latest version

File: maintainers/maintainer-list.nix
Comment (BhasherBEL) [reply maintainers/maintainer-list.nix:1]: I can split it if it's necessary

**Comment by BhasherBEL** [comment]
Feel free to merge the other PR, but it seems stale and is several versions behind.

**Comment by aikooo7** [comment]
> Feel free to merge the other PR, but it seems stale and was several versions behind. It also include the `Cargo.lock`, by the opposite to here.

1. It is in the same version as yours, wdym?
2. By my research it is kinda standard and happens In a lot of package to have the Cargo.lock, what is the issue?

**Comment by BhasherBEL** [comment]
>     1. Thanks for callout, I did destroy the commit that I had changed to use the latest version and didn't notice it, it should be the latest version now.

Yup, it's perfect now, my comment was anterior to that change.

>     2. You pull the Cargo.lock at build time, I pre-pulled it, both might exist and I am using what I saw in the rust's nixos documentation, what is the issue?

I can agree with you and did that at first, but I changed due to a [comment](https://github.com/NixOS/nixpkgs/pull/315864#discussion_r1633257386). 

To be honest I don't really care which PR is merged, but I would love to see this package in `nixpkgs`, so if it's faster with your PR, I'm in.

**Comment by BhasherBEL** [comment]
Closed in favor of https://github.com/NixOS/nixpkgs/pull/287858, which is now more up to date.
